frontend/javascripts/oxalis/geometries/materials/plane_material_factory.ts

```diff
@@ -399,6 +402,12 @@ class PlaneMaterialFactory {
     };
 
     this.material.side = THREE.DoubleSide;
+    this.startListening();
+
+    this.recomputeShaders();
```

Comment:
You are right, this is not ideal. I pushed a commit which works around this. Still not 100% satisfied with the solution, but I don't have a better idea for now..

Reply:
Sure, I see that the commit doesn't make this really clear and also was unnecessarily complex. I changed the code again so that basically the only remaining change is that the uniform listeners are set up before the initial shader code is generated. The reason for this is that one of the uniform listeners also mutates `this.scaledTpsInvPerLayer` and that property is necessary to compute the shader code. It's not super elegant, as I said before, but it does the trick :see_no_evil:

Reply:
Thanks, that makes sense :+1: